### DISCUSSION TITLE
Use kine to integrate with multiple data sources

### DISCUSSION THREAD

**Discussion by i18nsite** [apache/kvrocks-controller#146] [Ideas]
看这里文章
https://zhuanlan.zhihu.com/p/458705854
kine貌似是一个go库，封装各种数据库成为etcd 。
kvrocks-controller 如果调用这个库，
就不用自己搞各种数据库后端的封装了
![IMG_20240127_111349](https://github.com/apache/kvrocks-controller/assets/145643935/9e77292a-4c69-4eca-8d3f-cb630072b8e6)

![mmexport1706326229188](https://github.com/apache/kvrocks-controller/assets/145643935/d9c8aaa0-b812-40e2-821d-8921e5bc6cc8)

get
exist
set
delete
list
都有

**Comment by jihuayu** [comment]
Thanks for your suggestion. It seems like k3s use this lib(kine). I will have a look.
